src/intrinsic/simd.rs

```diff
@@ -722,15 +741,14 @@ pub fn generic_simd_intrinsic<'a, 'gcc, 'tcx>(bx: &mut Builder<'a, 'gcc, 'tcx>,
                         ),
                     }
 
-                    // boolean reductions operate on vectors of i1s:
-                    let i1 = bx.type_i1();
-                    let i1xn = bx.type_vector(i1, in_len as u64);
-                    bx.trunc(args[0].immediate(), i1xn)
+                    // TODO(antoyo): gcc doesn't like vectors of bools, so reduce as a full-wide
```

Comment:
I put it there because it's slightly different behavior from the llvm codegen backend, and at the time I wrote this, I wasn't sure how closely this was following the llvm backend.

Since libgccjit doesn't let you have a vector of bools, we have to do something different here (and in a few other places, particularly regarding comparisons and masking).

Removed in 664f705.